ehrapy/preprocessing/_encoding.py

```diff
@@ -502,94 +461,74 @@ def _update_encoded_data(
     var_names = [col_name for col_idx, col_name in enumerate(var_names) if col_idx not in idx]
     temp_var_names = categorical_prefixes + var_names
 
-    return temp_x, temp_var_names
+    unencoded_var_names = [col_name for col_idx, col_name in enumerate(unencoded_var_names) if col_idx not in idx]
+    unencoded_var_names = unencoded_prefixes + unencoded_var_names
+
+    return temp_x, temp_var_names, unencoded_var_names
 
 
 def _initial_encoding(
-    uns: dict[str, Any],
+    obs: pd.DataFrame,
     categoricals: list[str],
 ) -> np.ndarray:
     """Get all original values for all categoricals that need to be encoded (again).
 
     Args:
-        uns: A copy of the original AnnData object's uns
+        obs: A copy of the original obs where the original categorical values are stored that will be encoded
         categoricals: All categoricals that need to be encoded
 
     Returns:
         Numpy array of all original categorial values
     """
-    uns_: dict[str, np.ndarray] = uns
     # create numpy array from all original categorical values, that will be encoded (again)
-    array = np.array(
-        [uns_["original_values_categoricals"][categoricals[i]].ravel() for i in range(len(categoricals))]
-    ).transpose()
+    array = np.array([obs[categoricals[i]].ravel() for i in range(len(categoricals))]).transpose()
 
     return array
 
 
 def _undo_encoding(
     adata: AnnData,
-    columns: str = "all",
-    suppress_warning: bool = False,
+    verbose: bool = True,
 ) -> AnnData | None:
     """Undo the current encodings applied to all columns in X. This currently resets the AnnData object to its initial state.
 
     Args:
         adata: The AnnData object
-        columns: The names of the columns to reset encoding for. Defaults to all columns. This resets the AnnData object to its initial state.
-        suppress_warning: Whether warnings should be suppressed or not.
+        verbose: Set to False to suppress warnings. Defaults to True.
 
     Returns:
         A (partially) encoding reset AnnData object
     """
-    if "var_to_encoding" not in adata.uns.keys():
-        if not suppress_warning:
-            logger.warning("Calling undo_encoding on unencoded AnnData object.")
-        return None
+    if "encoding_mode" not in adata.var.keys():
+        if verbose:
+            logger.warning("Calling undo_encoding on unencoded AnnData object. Returning original AnnData object!")
```

Comment:
If I got this right, `undo_encoding` is dropped: so this message might be a bit confusing;
can it be dropped at this point, as the only case _undo_encoding is called is in encoding, after checking exactly

```py
if "encoding_mode" not in adata.var.keys():
```

Reply:
True, `encoding_mode` must be present in `adata.var` at this point. I removed it.